src/wallet/receive.cpp

```diff
@@ -304,17 +304,23 @@ Balance GetBalance(const CWallet& wallet, const int min_depth, bool avoid_reuse)
             const int tx_depth{wallet.GetTxDepthInMainChain(wtx)};
             const CAmount tx_credit_mine{CachedTxGetAvailableCredit(wallet, wtx, ISMINE_SPENDABLE | reuse_filter)};
             const CAmount tx_credit_watchonly{CachedTxGetAvailableCredit(wallet, wtx, ISMINE_WATCH_ONLY | reuse_filter)};
+            const CAmount tx_credit_mine_used{CachedTxGetAvailableCredit(wallet, wtx, ISMINE_SPENDABLE | ISMINE_USED)};
             if (is_trusted && tx_depth >= min_depth) {
                 ret.m_mine_trusted += tx_credit_mine;
                 ret.m_watchonly_trusted += tx_credit_watchonly;
             }
+            if (is_trusted) {
```

Comment:
This needs to be in the above `if` so that the min_depth check is still being done.

Reply:
https://github.com/bitcoin/bitcoin/blob/a46065e36cf868265c909dc5edf29dc17be53c1f/src/wallet/rpc/coins.cpp#L480

My understanding from the above line is `min_depth` should be `zero` when calculating `full_balance` hence I didn't include the statement below in the above `if` since it uses `min_depth` provided by the caller

I didn't include the check in this `if` since `tx_depth` is always >= 0

Reply:
I don't think it makes sense to return `m_mine_used` without respecting `min_depth`. The original code passes 0 for `min_depth` because that is the default value but it needs to pass the `avoid_reuse` flag. It is not because used balance should always calculated with a min depth of 0. Ignoring it is a layer violation here.